gradle-metric-schema/src/main/java/com/palantir/metric/schema/gradle/MetricSchemaPlugin.java

```diff
@@ -31,127 +33,90 @@
 import org.gradle.api.tasks.SourceSet;
 import org.gradle.api.tasks.TaskProvider;
 import org.gradle.plugins.ide.idea.model.IdeaModel;
-import org.gradle.plugins.ide.idea.model.IdeaModule;
 
 public final class MetricSchemaPlugin implements Plugin<Project> {
-    private static final String TASK_GROUP = "MetricSchema";
-    public static final String METRIC_SCHEMA_RESOURCE = "metric-schema/metrics.json";
 
-    public static final String COMPILE_METRIC_SCHEMA = "compileMetricSchema";
-    public static final String CHECK_METRICS_MARKDOWN = "checkMetricsMarkdown";
-    public static final String GENERATE_METRICS_MARKDOWN = "generateMetricsMarkdown";
-    public static final String CREATE_METRICS_MANIFEST = "createMetricsManifest";
-    public static final String GENERATE_METRICS = "generateMetrics";
+    public static final String TASK_GROUP = "MetricSchema";
+
+    public static final String METRICS_JSON_FILE = "metric-schema/metrics.json";
 
     @Override
     public void apply(Project project) {
         project.getPluginManager().apply(JavaLibraryPlugin.class);
-        SourceDirectorySet sourceSet = createSourceSet(project);
+
+        Provider<Directory> generatedJavaDir =
+                project.getLayout().getBuildDirectory().dir("generated/sources/metric-schema/java/main");
+        Provider<Directory> generatedResourcesDir =
+                project.getLayout().getBuildDirectory().dir("generated/sources/metric-schema/resources/main");
```

Comment:
Previously we would generate the `manifest.json` into the `build/metricSchema` directory. But `build/metricSchema` is used as the resource directory, so it's weird to add additional files in that directory after `processResources` has run.

It also preventing us from automatically wiring the resources directory up the to `compile` task because we couldn't use the entire directory as the output.

Resources are now put in `generated/sources/metric-schema/resources/main` and `metricSchema` is just use for metric-schema internal things like `manifest.json`.